### PR TITLE
carson and max c

### PR DESCRIPTION
**Pull request by Ishan1522**

### CODE REVIEW

**Review by Ishan1522 — COMMENTED** [review]
there are more things here but here are some stuffs to look over.

File: src/main/java/frc/robot/commands/autodrive/AutoAlignWithAmp.java [hunk 1]
```diff
@@ -23,24 +23,24 @@ public class AutoAlignWithAmp extends Command {
 
   private final ProfiledPIDController turnController =
       new ProfiledPIDController(
-          ShooterConstants.AUTO_LINEUP_ROTATION_P,
-          ShooterConstants.AUTO_LINEUP_ROTATION_I,
-          ShooterConstants.AUTO_LINEUP_ROTATION_D,
-          ShooterConstants.AUTO_LINEUP_ROTATION_CONSTRAINTS);
+          FlywheelConstants.AUTO_LINEUP_ROTATION_P,
```
Comment (JacksonElia) [inline src/main/java/frc/robot/commands/autodrive/AutoAlignWithAmp.java:26]: why did we rename this? I think `ShooterConstants` was more accurate, especially for this case. The auto lineup stuff doesn't have anything to do with specifically the flywheel

Comment (Ishan1522) [reply src/main/java/frc/robot/commands/autodrive/AutoAlignWithAmp.java:26]: it was renamed to match the subsystem name. imo auto lineup constants should actually go in trajectory constants bc the shooter isn't the thing thats lining up w/ the amp, the swerve is.

Comment (JacksonElia) [reply src/main/java/frc/robot/commands/autodrive/AutoAlignWithAmp.java:26]: yeah we should move it to trajectory constants, also consider renaming trajectory constants to auto constants

File: src/main/java/frc/robot/subsystems/shooter/Flywheel.java [hunk 2]
```diff
@@ -47,13 +51,17 @@ public void setRollerSpeed(double speed) {
     Logger.recordOutput("Roller/DutyCycleOut", speed);
   }
 
+  /** updates and logs inputs periodically */
```
Comment (JacksonElia) [inline src/main/java/frc/robot/subsystems/shooter/Flywheel.java:54]: methods with @Override shouldn't be given docstrings, they already have them (try hovering over them)

File: src/main/java/frc/robot/subsystems/shooter/FlywheelConstants.java [hunk 3]
```diff
@@ -4,17 +4,33 @@
...
   public static final int ROLLER_MOTOR_ID = 2;
 
+  /** Sets the limit for the shooter supply voltage */
   public static final double SHOOTER_SUPPLY_LIMIT = 60;
+
+  /** Sets the limit for the shooter stator voltage */
   public static final double SHOOTER_STATOR_LIMIT = 60;
+
+  /** Enables the limit for stator */
   public static final boolean SHOOTER_STATOR_ENABLE = true;
```
[the remark this answers is not in the document]
Comment (Ishan1522) [reply src/main/java/frc/robot/subsystems/shooter/FlywheelConstants.java:25]: just put a link to this doc : https://v6.docs.ctr-electronics.com/en/stable/docs/hardware-reference/talonfx/improving-performance-with-current-limits.html

File: src/main/java/frc/robot/subsystems/shooter/FlywheelIOTalonFX.java [hunk 4]
```diff
@@ -90,18 +91,15 @@ public FlywheelIOTalonFX() { // Object to set different flywheel configs
...
   public void updateInputs(
       FlywheelIOInputs inputs) { // gets current values for motors and puts them into a log
     BaseStatusSignal.refreshAll(
         leaderPosition, leaderVelocity, leaderAppliedVolts, leaderCurrent, followerCurrent);
     inputs.positionRotations =
         leaderPosition.getValueAsDouble()
-            / ShooterConstants.GEAR_RATIO; // converted to radians to gear ratio math
-    inputs.velocityRPM = (leaderVelocity.getValueAsDouble() * 60.0) / ShooterConstants.GEAR_RATIO;
+            / FlywheelConstants.GEAR_RATIO; // converted to radians to gear ratio math
+    inputs.velocityRPM = (leaderVelocity.getValueAsDouble() * 60.0) / FlywheelConstants.GEAR_RATIO;
```
Comment (JacksonElia) [inline src/main/java/frc/robot/subsystems/shooter/FlywheelIOTalonFX.java:102]: I think this * 60.0 should be a constant, you can put it in HardwareConstants or something

**Comment by Ishan1522** [comment]
its been so long i forgot there was a pr open for this 💀

**Comment by Ishan1522** [comment]
@mychenezpz please make the changes requested in the pull request

File: src/main/java/frc/robot/subsystems/shooter/FlywheelConstants.java [hunk 5]
```diff
@@ -4,17 +4,33 @@
...
+
+  /** Enables the limit for supply */
```
[the remark this answers is not in the document]
Comment (Ishan1522) [reply src/main/java/frc/robot/subsystems/shooter/FlywheelConstants.java:27]: Done like ezpz

File: src/main/java/frc/robot/subsystems/shooter/FlywheelConstants.java [hunk 6]
```diff
@@ -24,11 +40,22 @@ public class ShooterConstants {
 
   public static final int SHOOTER_ACCEPTABLE_RPM_ERROR = 25;
 
+  /** Forced Labor has determined that this is the best P value */
   public static final double SHOOT_P = 0.522;
```
[the remark this answers is not in the document]
Comment (Ishan1522) [reply src/main/java/frc/robot/subsystems/shooter/FlywheelConstants.java:44]: Resolved ezpzz

File: src/main/java/frc/robot/commands/autodrive/AutoAlignWithAmp.java (hunk 1 above)
Comment (mychenezpz) [reply src/main/java/frc/robot/commands/autodrive/AutoAlignWithAmp.java:26]: Looks like im right so Im keeping it as flywheel

**Review by mychenezpz — APPROVED** [review]
> I think this * 60.0 should be a constant, you can put it in HardwareConstants or something 

OK sure

File: src/main/java/frc/robot/subsystems/shooter/FlywheelConstants.java [hunk 7]
```diff
@@ -24,11 +40,22 @@ public class ShooterConstants {
...
+
+  /** Forced Labor has determined that this is the best I value */
```
[the remark this answers is not in the document]
Comment (mychenezpz) [reply src/main/java/frc/robot/subsystems/shooter/FlywheelConstants.java:46]: No

File: src/main/java/frc/robot/subsystems/shooter/FlywheelIOTalonFX.java [hunk 8]
```diff
@@ -90,18 +91,15 @@ public FlywheelIOTalonFX() { // Object to set different flywheel configs
...
   @Override
   public void updateInputs(
       FlywheelIOInputs inputs) { // gets current values for motors and puts them into a log
     BaseStatusSignal.refreshAll(
         leaderPosition, leaderVelocity, leaderAppliedVolts, leaderCurrent, followerCurrent);
     inputs.positionRotations =
         leaderPosition.getValueAsDouble()
-            / ShooterConstants.GEAR_RATIO; // converted to radians to gear ratio math
-    inputs.velocityRPM = (leaderVelocity.getValueAsDouble() * 60.0) / ShooterConstants.GEAR_RATIO;
+            / FlywheelConstants.GEAR_RATIO; // converted to radians to gear ratio math
```
[the remark this answers is not in the document]
Comment (mychenezpz) [reply src/main/java/frc/robot/subsystems/shooter/FlywheelIOTalonFX.java]: ok removed

File: src/main/java/frc/robot/subsystems/shooter/FlywheelIOTalonFX.java (hunk 4 above)
Comment (mychenezpz) [reply src/main/java/frc/robot/subsystems/shooter/FlywheelIOTalonFX.java]: ok

File: src/main/java/frc/robot/subsystems/shooter/FlywheelIO.java [hunk 9]
```diff
@@ -21,13 +22,17 @@ public default void setVoltage(double volts) {}
   /**
    * Run closed loop at the specified velocity.
    *
-   * @param velocityRPM Recieve desired input in rounds per minute, and the method will convert to
-   *     RPS to match requirements for VelocityVoltage
+   * @param velocityRPM Recieve desired input in rotations per minute
    */
   public default void setVelocity(double velocityRPM) {}
 
   /** Stop in open loop. */
```
[the remark this answers is not in the document]
Comment (mychenezpz) [reply src/main/java/frc/robot/subsystems/shooter/FlywheelIO.java:29]: idk how it works so Idk ig idk so idk idk idk idk. Senior Max wrote this so ask him.

File: src/main/java/frc/robot/subsystems/shooter/Flywheel.java [hunk 10]
```diff
@@ -7,19 +7,18 @@ public class Flywheel extends SubsystemBase {
   private final FlywheelIO io;
   private final FlywheelIOInputsAutoLogged inputs = new FlywheelIOInputsAutoLogged();
 
-  // private final SimpleMotorFeedforward ffModel;
-
+  /** Constructor that creates initiates IO */
```
[the remark this answers is not in the document]
Comment (mychenezpz) [reply src/main/java/frc/robot/subsystems/shooter/Flywheel.java:10]: ok

File: src/main/java/frc/robot/subsystems/shooter/Flywheel.java [hunk 11]
```diff
@@ -33,6 +32,11 @@ public void setFlywheelVelocity(double desiredRPM) {
     Logger.recordOutput("Flywheel/RPM", desiredRPM);
   }
 
+  /**
+   * Returns whether or not a note is detected
+   *
+   * @return whether a note was detected or not
```
[the remark this answers is not in the document]
Comment (mychenezpz) [reply src/main/java/frc/robot/subsystems/shooter/Flywheel.java:38]: ok hopefully this is right I think Max inverted the sensor so it be like this

File: src/main/java/frc/robot/subsystems/shooter/Flywheel.java (hunk 2 above)
Comment (mychenezpz) [reply src/main/java/frc/robot/subsystems/shooter/Flywheel.java]: Ok Ishan told me to docstring for override

File: src/main/java/frc/robot/subsystems/shooter/Flywheel.java [hunk 12]
```diff
@@ -47,13 +51,17 @@ public void setRollerSpeed(double speed) {
     Logger.recordOutput("Roller/DutyCycleOut", speed);
   }
 
+  /** updates and logs inputs periodically */
   @Override
   public void periodic() {
     // This method will be called once per scheduler run
     io.updateInputs(inputs);
     Logger.processInputs("FlywheelSubsystem", inputs);
```
[the remark this answers is not in the document]
Comment (mychenezpz) [reply src/main/java/frc/robot/subsystems/shooter/Flywheel.java:57]: ok matched

File: src/main/java/frc/robot/subsystems/shooter/Flywheel.java [hunk 13]
```diff
@@ -47,13 +51,17 @@ public void setRollerSpeed(double speed) {
...
   }
 
+  /**
+   * @return periodic but for the simulation
```
[the remark this answers is not in the document]
Comment (mychenezpz) [reply src/main/java/frc/robot/subsystems/shooter/Flywheel.java:61]: Doesnt even matter; Jack said that @override alr has it

File: src/main/java/frc/robot/subsystems/shooter/FlywheelConstants.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (mychenezpz) [reply src/main/java/frc/robot/subsystems/shooter/FlywheelConstants.java:27]: Idk what it supposed to be tbh

File: src/main/java/frc/robot/subsystems/shooter/FlywheelIOTalonFX.java [hunk 14]
```diff
@@ -41,23 +40,25 @@ public class FlywheelIOTalonFX
   public FlywheelIOTalonFX() { // Object to set different flywheel configs
     TalonFXConfiguration flywheelConfig = new TalonFXConfiguration();
     flywheelConfig.CurrentLimits.SupplyCurrentLimit =
-        ShooterConstants
+        FlywheelConstants
             .FLYWHEEL_SUPPLY_LIMIT; // Talonfx configuration software limits found in CONSTANTS file
```
[the remark this answers is not in the document]
Comment (mychenezpz) [reply src/main/java/frc/robot/subsystems/shooter/FlywheelIOTalonFX.java]: DONE

File: src/main/java/frc/robot/commands/autodrive/AutoAlignWithAmp.java (hunk 1 above)
Comment (Ishan1522) [reply src/main/java/frc/robot/commands/autodrive/AutoAlignWithAmp.java]: What

Comment (Ishan1522) [reply src/main/java/frc/robot/commands/autodrive/AutoAlignWithAmp.java]: Bro did not read the conversation

Comment (Ishan1522) [reply src/main/java/frc/robot/commands/autodrive/AutoAlignWithAmp.java]: Bro did not read the conversation

**Pull request merged**